tensorpack/tfutils/argscope.py

```diff
@@ -60,3 +62,30 @@ def get_arg_scope():
         return _ArgScopeStack[-1]
     else:
         return defaultdict(dict)
+
+
+def tflayers_mapper(func):
+    """Decorator for tf.layers.method to support argscope
+    """
+    @wraps(func)
+    def wrapped_func(*args, **kwargs):
+        actual_args = copy.copy(get_arg_scope()[func.__name__])
+        actual_args.update(kwargs)
+        out_tensor = func(*args, **actual_args)
+        print('passed kwargs', actual_args)
+        return out_tensor
+    a = wrapped_func
+    a.symbolic_function = None
+    return a
+
+
+def patch_tflayers(module, decorator=tflayers_mapper):
```

Comment:
Maybe don't need to pass in the module, since this function is written for `tf.layers` only

Reply:
I renamed this function as it is quite general.